### PR TITLE
Vendor a real-world GCN message-passing model as a multi-module fixture

### PR DESCRIPTION
**Pull request by khatchad**
Follow-up to #368, broadening tensor-type inference coverage with a multi-module real-world fixture.

Main change: a faithful **multi-file** fixture and parameter-type test for `GCNLayer.call`, a real-world graph-neural-network model from `kyzhouhzau/NLPGNN`. The `GCNLayer`, `GraphConvolution`, and `MessagePassing` modules are vendored verbatim under a package layout (`gcn_proj/nlpgnn/...`); only a driver and a reachable-slice `nlpgnn/gnn/utils.py` (just the `GNNInput` named tuple) are bespoke. This exercises inference across a four-module import chain (`driver→GCN→GCNConv→MessagePassing`), following the existing multi-module test convention (`testModule14` and friends).

- The tensor parameter `node_embeddings` is recovered concretely as `(4, 8) float32`, flowing across the module boundaries through `tf.keras.Model.__call__` dispatch. (`adjacency_lists` is a Python list of edge tensors, not a tensor itself, so it is not a tensor parameter.)
- The message-passing output locals are ⊤ on both axes: the custom `tf.scatter_nd` / `tf.math.unsorted_segment_sum` aggregation inside `MessagePassing` is not modeled, so the layer output type is lost. The decorated function's input signature is nonetheless exact.

Also included:
- A comment in the `evaluate` fixture noting that the upstream `tf.sqrt(..., tf.float32)` name-arg quirk and scalar normalization are reproduced verbatim from the analyzed source, not transcription errors (per the #368 review thread).
- A Javadoc touch-up dropping spaces around em-dashes and arrows in the batch-2 fixture tests.

The driver runs to completion under `python3.10` with `assert`s on the runtime shape and dtype matching the static-analysis expectations.

### CODE REVIEW

File: com.ibm.wala.cast.python.test/data/gcn_proj/nlpgnn/gnn/messagepassing.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        self, message_per_type, edge_type_to_message_targets, num_nodes
+    ):
+        if self.aggr == "sum":
+            aggregation_fn = tf.math.unsorted_segment_sum
+        elif self.aggr == "max":
+            aggregation_fn = tf.math.unsorted_segment_max
+        elif self.aggr == "mean":
+            aggregation_fn = tf.math.unsorted_segment_mean
+        elif self.aggr == "sqrt":
+            aggregation_fn = tf.math.unsorted_segment_sqrt_n
```
[the remark this answers is not in the document]
Comment (khatchad) [reply com.ibm.wala.cast.python.test/data/gcn_proj/nlpgnn/gnn/messagepassing.py:155]: > `MessagePassing.__init__` defaults `aggr` to "add", but `_aggregate_function` doesn't handle "add" [...] will leave `aggregation_fn` undefined and raise `UnboundLocalError`

This module is vendored verbatim from upstream `kyzhouhzau/NLPGNN` for analysis fidelity, so the latent inconsistency is upstream's rather than something to patch here. It is also not reachable on this fixture's path: `GraphConvolution.__init__` overrides the default to `aggr="sum"`, and `GCNLayer` constructs both layers with that default, so `_aggregate_function` takes the defined `"sum"` branch. The driver runs to completion under `python3.10`.

File: com.ibm.wala.cast.python.test/data/gcn_proj/nlpgnn/gnn/GCNConv.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        bias_initializer="zeros",
+        regularizer=None,
+        use_bias=False,
+        cached=True,
+        **kwargs
+    ):
+        super(GraphConvolution, self).__init__(aggr, **kwargs)
+        self.kernel_initializer = tf.keras.initializers.get(kernel_initializer)
+        self.bias_initializer = tf.keras.initializers.get(bias_initializer)
+        self.regularizer = tf.keras.initializers.get(regularizer)
```
[the remark this answers is not in the document]
Comment (khatchad) [reply com.ibm.wala.cast.python.test/data/gcn_proj/nlpgnn/gnn/GCNConv.py:28]: > `regularizer` is retrieved via `tf.keras.initializers.get` [...] should use `tf.keras.regularizers.get`

Same situation: this is upstream `kyzhouhzau/NLPGNN` code vendored verbatim, so the type confusion is upstream's. It is also inert on this fixture's path: the default `regularizer=None` makes `tf.keras.initializers.get(None)` return `None`, and `use_bias=False`, so `add_weight` receives no regularizer. The driver runs clean under `python3.10`.
